src/js/components/AppListItemLabelsComponent.jsx

```diff
@@ -0,0 +1,159 @@
+var classNames = require("classnames");
+var React = require("react/addons");
+
+var OnClickOutsideMixin = require("react-onclickoutside");
+var Util = require("../helpers/Util");
+
+var AppListItemLabelsComponent = React.createClass({
+  displayName: "AppListItemLabelsComponent",
+
+  initialMarginTop: null,
```

Comment:
Thanks for the feedback! Refactored to make use of the "shared" object with reactid keys.